### PR TITLE
fix: Add function to store data from CA post calls.

### PR DESCRIPTION
**Pull request by rafiu007**

### CODE REVIEW

File: bayesian/utils.py [hunk 1]
```diff
@@ -64,16 +63,19 @@ def get_user_email(user_profile):
         return default_email
 
 
-def server_create_component_bookkeeping(ecosystem, name, version, user_profile):
+def create_component_bookkeeping(ecosystem, packages_list,
+                                 user_id, user_agent, manifest_hash, request_id):
     """Run the component analysis for given ecosystem+package+version."""
     args = {
-        'external_request_id': uuid.uuid4().hex,
         'data': {
```
Comment (prashbnair) [inline bayesian/utils.py:70]: Why are the other fields not in the data dict?

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: bayesian/utils.py (hunk 1 above)
Comment (rafiu007) [reply bayesian/utils.py:70]: No particular reason.I ll move it.

**Review by prashbnair — APPROVED** [review]

File: bayesian/api/api_v2.py [hunk 2]
```diff
@@ -199,6 +197,9 @@ def post():
         response_template: Tuple = namedtuple("response_template", ["message", "status", "headers"])
         input_json: Dict = request.get_json()
         ecosystem: str = input_json.get('ecosystem')
+        user_agent = request.headers.get('User-Agent', None)
+        manifest_hash = str(request.headers.get('manifest_hash', None))
```
Comment (arajkumar) [inline bayesian/api/api_v2.py:201]: Why are we overloading request header? shouldn't this be optional filed of request payload?

File: bayesian/api/api_v2.py [hunk 3]
```diff
@@ -199,6 +197,9 @@ def post():
...
+        request_id = request.headers.get('request_id', None)
```
Comment (arajkumar) [inline bayesian/api/api_v2.py:202]: Why are we overloading request header? shouldn't this be optional filed of request payload?

File: bayesian/utils.py [hunk 4]
```diff
@@ -64,15 +63,18 @@ def get_user_email(user_profile):
...
     args = {
-        'external_request_id': uuid.uuid4().hex,
+        'external_request_id': request_id,
         'data': {
-            'api_name': 'component_analyses',
-            'user_email': get_user_email(user_profile),
-            'user_profile': user_profile,
-            'request': {'ecosystem': ecosystem, 'name': name, 'version': version}
+            'api_name': 'component_analyses_post',
+            'manifest_hash': hashlib.md5(manifest_hash.encode()).hexdigest(),
```
Comment (arajkumar) [inline bayesian/utils.py:73]: The field name in the request says manifest_hash, then why are computing again here?

Comment (rafiu007) [reply bayesian/utils.py:73]: Manifeat_hash will actually contain the file path.

File: bayesian/api/api_v2.py (hunk 3 above)
Comment (rafiu007) [reply bayesian/api/api_v2.py:202]: @prashbnair

File: bayesian/utils.py (hunk 1 above)
Comment (rafiu007) [reply bayesian/utils.py]: Resolved.

File: bayesian/utils.py (hunk 4 above)
Comment (rafiu007) [reply bayesian/utils.py:73]: when in the header.

File: bayesian/api/api_v2.py (hunk 3 above)
Comment (prashbnair) [reply bayesian/api/api_v2.py:202]: I dont think any of these fields are optional. I am now having a doubt if we should capture the path as it reveals some data about the user. I read somewhere that it is not allowed.

Comment (arajkumar) [reply bayesian/api/api_v2.py:202]: we can apply the hash at client level. however still SA request has manifest path.

File: bayesian/utils.py (hunk 4 above)
Comment (arajkumar) [reply bayesian/utils.py:73]: it should be fixed.

also it needs api spec change

**Pull request merged**